inference-engine/src/mkldnn_plugin/nodes/extract_image_patches.cpp

```diff
@@ -0,0 +1,242 @@
+            auto rates = layer->GetParamAsUInts("rates", {1u, 1u});
+            _padding = layer->GetParamAsString("auto_pad", "valid");
+
+            _ksizes.clear();
+            _strides.clear();
+            _rates.clear();
+            for (size_t i = 0; i < ksizes.size(); i++)
+                _ksizes.push_back((int64_t)ksizes[i]);
+            for (size_t i = 0; i < strides.size(); i++)
+                _strides.push_back((int64_t)strides[i]);
+            for (size_t i = 0; i < rates.size(); i++)
+                _rates.push_back((int64_t)rates[i]);
+
+            LayerConfig config;
+
+            DataConfig inConfig;
+            inConfig.desc = inData->getTensorDesc();
+            config.inConfs.push_back(inConfig);
+
+            DataConfig outConfig;
+            outConfig.desc = layer->outData[0]->getTensorDesc();
+            config.outConfs.push_back(outConfig);
+
+            config.dynBatchSupport = false;
+            confs.push_back(config);
+        } catch (InferenceEngine::details::InferenceEngineException &ex) {
+            errorMsg = ex.what();
+        }
+    }
+
+    StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept override {
+        switch (inputs[0]->getTensorDesc().getPrecision().size()) {
+            case 1: {
+                process_data<PrecisionTrait<Precision::U8>::value_type>(inputs, outputs);
+                break;
+            }
+            case 2: {
+                process_data<PrecisionTrait<Precision::U16>::value_type>(inputs, outputs);
+                break;
+            }
+            case 4: {
+                process_data<PrecisionTrait<Precision::I32>::value_type>(inputs, outputs);
+                break;
+            }
+            case 8: {
+                process_data<PrecisionTrait<Precision::U64>::value_type>(inputs, outputs);
+                break;
+            }
+            default: {
+                if (resp) {
+                    std::string errorMsg = "ExtractImagePatches layer does not support precision '"
+                            + std::string(inputs[0]->getTensorDesc().getPrecision().name()) + "'";
+                    errorMsg.copy(resp->msg, sizeof(resp->msg) - 1);
+                }
+                return GENERAL_ERROR;
+            }
+        }
+
+        return OK;
+    }
+
+    template<typename T>
+    void process_data(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs) noexcept {
+        const T* src_data = inputs[0]->cbuffer().as<const T*>() +
+            inputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        T* dst_data = outputs[0]->buffer().as<T*>() +
+            outputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+
+        const auto& inDims = inputs[0]->getTensorDesc().getDims();
+        const size_t inDimsSize = inDims.size();
+
+        const size_t BATCH = 0, CHANNEL = 1,
+            HIGHT = 0, WIDTH = 1;
```

Comment:
What's the reason this line was partially moved to a new line?

Reply:
Fixed